tests/istio/extension_reconciler_test.go

```diff
@@ -316,6 +317,56 @@ var _ = Describe("Rate Limiting WasmPlugin controller", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Eventually(tests.RouteIsAccepted(ctx, testClient(), client.ObjectKeyFromObject(httpRoute))).WithContext(ctx).Should(BeTrue())
 
+			//create authpolicy
```

Comment:
I wonder if a more rounded test to also test the unhappy path also would be to move the creation of this authpolicy to **after** the rlp created at https://github.com/Kuadrant/kuadrant-operator/blob/d43c0910d9a6a6974ecba0d5f6177aa7c6a5b3b3/tests/istio/extension_reconciler_test.go#L414

Before creating the authpolicy, we can test that the RLP is not enforced with a CEL validation error, and that no `WasmPlugin` CR/config was created 🤔 

Once the authpolicy is created, everything will be in place and RLP will be enforced and WasmPlugin config is created.

This would essentially test the verification steps described in the PR 🤔